doc/manual/rl-next/source-positions-in-errors.md

```diff
@@ -0,0 +1,45 @@
+synopsis: Source locations are printed more consistently in errors
+issues: #561
+prs: #9555
+description: {
+
+Source location information is now included in error messages more
+consistently. Given this code:
+
+```nix
+let
+  attr = {foo = "bar";};
+  key = {};
+in
+  attr.${key}
+```
+
+Previously, Nix would show this unhelpful message when attempting to evaluate
+it:
+
+```
+error:
+       … while evaluating an attribute name
+
+         at «none»:0: (source not available)
```

Comment:
```suggestion
```

It didn't do that anymore since 2.19.
- https://github.com/NixOS/nix/pull/8905

You may set `_NIX_DEVELOPER_SHOW_UNKNOWN_LOCATIONS=1` to show the missing positions once more.